### ISSUE TITLE
Rotate in version 0.5.1

### ISSUE THREAD

**Issue by MichaPau** [processing/p5.js#1465]
rotate() seems to be broken in version 0.5.1. It just doesn't rotate any more.

Simple [codePen example](http://codepen.io/anon/pen/MeyKLx?editors=1010)

In this example 2 versions are included. Version 0.5.1 and 0.4.24. The version 0.4.24 canvas shows the line rotated, the version 0.5.1 canvas doesn't. (in Firefox 47 and Chrome 51)

processing code in this example:

```
function setup() {
  createCanvas(300, 300);
  
  background(0);
  stroke("#FFFFFF");
  translate(width/2, height/2);
 	rotate(radians(45));
  line(-50, 0, 50, 0);
}

function draw() {
  
}
```

**Comment by BenjaminHabert** [comment]
I experienced a similar issue.

**Comment by lmccart** [comment]
the matrix is now reset at the beginning of draw, so the best thing to do would be to move the `rotate(radians(45))` line to the beginning of the draw loop.

**Comment by MichaPau** [comment]
Same result when moving rotate to the draw() function.
```
function setup() {
  createCanvas(300, 300);
}

function draw() {
  background(0);
  stroke("#FFFFFF");
  translate(width/2, height/2);
  rotate(radians(45));
  line(-50, 0, 50, 0);
}
```

**Comment by lmccart** [comment]
ah ok, thanks! I will take a look.

**Comment by danazkari** [comment]
+1 I could help taking a look into this later on today as well, I like this lib very much and would be more than glad to contribute :)

**Comment by lmccart** [comment]
hey @danazkari if you want to take a look into this, that'd be great!

**Comment by danazkari** [comment]
Ok so I think I found the little rascal... and also found a few small optimizations so I'll be posting a PR pretty soon hopefully!

@lmccart do you have a preferred way to call the branches for the fixes or should I just go crazy with the naming? :wink:

**Comment by lmccart** [comment]
go crazy! :)

**Comment by danazkari** [comment]
Hi @lmccart, so.. I'm trying to push a new branch I created based on master and it throws me a 403, do I need to have some special permissions for me to push my branch?

BTW, the optimization couldn't happen because of browser incompatibility. But I did the fix which is the important bit.

**Comment by lmccart** [comment]
Ah, rather than pushing a branch, could you [submit a pull request](https://help.github.com/articles/using-pull-requests/) from the fix branch in your own repository to the master branch of processing's version of the repository?

**Comment by lmccart** [comment]
If you haven't done so already, you may need to fork this repo into your own github account and then copy in the changes you've made to a branch there.

**Comment by danazkari** [comment]
@lmccart thanks a ton! That helped :) sorry, I'm a newbie contributing to open source projects in GitHub.

Anyways, here https://github.com/processing/p5.js/pull/1466 lemme know if it needs something else.

**Comment by lmccart** [comment]
cool, this is fixed with #1466. and @MichaPau I was mistaken, it's fine to put rotate() in setup, it doesn't need to be in draw(). should work with the fix in the next release.
